rust/src/quic/quic.rs

```diff
@@ -0,0 +1,336 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use super::{
+    cyu::Cyu,
+    parser::{QuicData, QuicHeader},
+};
+use crate::applayer::{self, *};
+use crate::core::{self, AppProto, Flow, ALPROTO_FAILED, ALPROTO_UNKNOWN, IPPROTO_UDP};
+use std::ffi::CString;
+use std::mem::transmute;
```

Comment:
I think `transmute` got removed from Suricata, @jasonish did you do it ?

Reply:
(should it be added to cocci checks of forbidden functions ?)

Reply:
Yeah, all transmutes have been replaced with costs or Box::into_raw,from_raw.